lib/internal/per_context/primordials.js

```diff
@@ -12,17 +12,8 @@
 // `primordials.Object` where `primordials` is a lexical variable passed
 // by the native module compiler.
 
-const ReflectApply = Reflect.apply;
-
-// This function is borrowed from the function with the same name on V8 Extras'
-// `utils` object. V8 implements Reflect.apply very efficiently in conjunction
-// with the spread syntax, such that no additional special case is needed for
-// function calls w/o arguments.
-// Refs: https://github.com/v8/v8/blob/d6ead37d265d7215cf9c5f768f279e21bd170212/src/js/prologue.js#L152-L156
-function uncurryThis(func) {
-  return (thisArg, ...args) => ReflectApply(func, thisArg, args);
-}
-
+const { bind, call } = Function.prototype;
+const uncurryThis = call.bind(bind, call);
```

Comment:
not that this isn't a very cool application, but wouldn't this be a bit clearer?
```js
const uncurryThis = (f) => Function.prototype.call.bind(f)
```

Reply:
The issue is that `Function.prototype.call` can be modified by the user. I could add a comment explaining it though. Do you think that would be enough?

Reply:
Ah yeah I guess once you factor it out you just end up with `call.bind(bind, call)`. A comment would be good then.

Reply:
Done, PTAL.